### PR TITLE
Add code challenge methods for oidc provider configuration response

### PR DESCRIPTION
**Pull request by drunkcattt**
Add code challenge method for oidc well-known configuration endpoint.

Closes gh-1302

### CODE REVIEW

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java [hunk 1]
```diff
@@ -111,7 +111,8 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 				.tokenIntrospectionEndpointAuthenticationMethods(clientAuthenticationMethods())
 				.subjectType("public")
 				.idTokenSigningAlgorithm(SignatureAlgorithm.RS256.getName())
-				.scope(OidcScopes.OPENID);
+				.scope(OidcScopes.OPENID)
+				.codeChallengeMethod("S256");
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:115]: @drunkcattt Can you please update the test in `OidcProviderConfigurationEndpointFilterTests` similar to `OAuth2AuthorizationServerMetadataEndpointFilterTests`.

Comment (jgrandja) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:115]: @drunkcattt Are you able to update the test in `OidcProviderConfigurationEndpointFilterTests`?

Comment (drunkcattt) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/web/OidcProviderConfigurationEndpointFilter.java:115]: It's been added.

**Comment by jgrandja** [comment]
Thanks for the PR @drunkcattt. This is now merged.

**Pull request closed without merging**